sympy/combinatorics/pc_groups.py

```diff
@@ -0,0 +1,67 @@
+from sympy.combinatorics.fp_groups import FpGroup, FpSubgroup, subgroup_quotient
+from sympy import sieve
+
+class PcGroup(FpGroup):
+    '''
+    Class implementing Polycyclic groups.
+    To-do : Docstring completion.
+            Add error and exception handling
+    '''
+    def __init__(fp_group, pc_series=None, pcgs=None):
```

Comment:
I think that the definition should be different. I would expect something like a method `pc_presentation` in the `FpGroup` class that checks if the group is polycyclic first, then computes `pc_series` and only then returns a `PcGroup` presentation. And a polycyclic presentation should have different generators and relators from the original group anyway, so perhaps `__init__` could just take in `pc_series` as an argument, compute the polycyclic sequence to use as generators, compute the right relators, and only then return an instance of `PcGroup`.

Another possibility is doing what's  described in the Handbook in Definition 8.7, when a polycyclic presentation is first defined. You could take in `S` as an argument and build up a polycyclic group from that. This would also mean that you could test your polycyclic methods before you sort out obtaining a polycyclic series for an `FpGroup`.

Reply:
> perhaps __init__ could just take in pc_series as an argument, compute the polycyclic sequence to use as generators, compute the right relators, and only then return an instance of PcGroup.

Well, I haven't pushed it yet, but I've changed the definition a bit. Currently, I've written it as `__init__(self, free_group, pc_series=None, pcgs=None, relations=None)`. I've set `pc_series`, `pcgs`, `relations` as default arguments because the user could either choose to specify the `pc_series` directly or he could define a presenation. And the `free_group` argument would be needed anyway.

Reply:
Why would you need `free_group` as an argument? You can just build one that has the same length as the given `pc_series`, and this what you'd have to do if `free_group` is the wrong length. And what would happen if the user just specified a `free_group` and none of the other arguments? It seems like there could be problems with it.

Another issue with letting a user specify a presentation is that they could specify something that isn't actually polycyclic. While if we enforced that presentation have to be specified with `S`, then we never run into that problem